tests/collections/llm/gpt_finetuning.py

```diff
@@ -55,6 +56,8 @@ def get_args():
     strategy = nl.MegatronStrategy(
         tensor_model_parallel_size=args.tp_size,
         pipeline_model_parallel_size=args.pp_size,
+        # Pipeline dtype is coupled with the bf16 mixed precision plugin
+        pipeline_dtype=torch.bfloat16,
```

Comment:
fyi @hemildesai @BoxiangW - this requirement is coming after https://github.com/NVIDIA/NeMo/pull/10954  
by the way the validation logic is written, users are required to set this both on the strategy directly, and possibly optionally through a precision plugin passed to the strategy or trainer. this duplication feels wrong to do

Reply:
I think a better way than hardcoding `pipeline_dtype` is to make it a function attribute and set it's value if it's used https://github.com/NVIDIA/NeMo/pull/11504/files#diff-78f81f4094cfea056c177e87c0d527b9ce27cee11813138e5a2a69370b922c19R282